File: mobile/ios/Sources/OpenSourceOrthoLiteKit/EngineConfig.swift
import Foundation

/// Where the lite app reaches the OpenSource Ortho Python engine.
///
/// The engine is the single source of truth (see ../../API_CONTRACT.md); the app
/// never synthesizes a plan on-device. Change `baseURL` in one place to point at
/// a deployed engine. Cleartext HTTP is only for the local dev hosts below.
public struct EngineConfig: Sendable {
    public var baseURL: URL

    public init(baseURL: URL) {
        self.baseURL = baseURL
    }

    /// iOS Simulator reaches the developer's host loopback directly.
    public static let simulator = EngineConfig(baseURL: URL(string: "http://127.0.0.1:8000")!)

    /// Default used by the app shell. Swap for an https engine in production.
    public static let `default` = EngineConfig.simulator
}

File: mobile/android/app/src/main/kotlin/com/opensourceortho/lite/EngineConfig.kt
package com.opensourceortho.lite

/**
 * Where the lite app reaches the OpenSource Ortho Python engine.
 *
 * The engine is the single source of truth (see ../../API_CONTRACT.md); the app
 * never synthesizes a plan on-device. Change [baseUrl] in one place to point at a
 * deployed engine. Cleartext HTTP is only allowed for the dev hosts in
 * res/xml/network_security_config.xml.
 */
data class EngineConfig(val baseUrl: String) {
    companion object {
        /** Android emulator reaches the developer's host loopback via 10.0.2.2. */
        val EMULATOR = EngineConfig("http://10.0.2.2:8000")

        /** Default used by the app shell. Swap for an https engine in production. */
        val DEFAULT = EMULATOR
    }
}
